### PR TITLE
feat: Add Role to manage Zesty secret

### PR DESCRIPTION
**Pull request by bartosz-qubex**

### CODE REVIEW

File: charts/zesty/templates/deployment.yaml [hunk 1]
```diff
@@ -83,6 +83,10 @@ spec:
           {{- end }}
           {{- if ne .Values.encryptedCredentials ""}}
           env:
+            - name: NAMESPACE
```
Comment (IdoKendo) [inline charts/zesty/templates/deployment.yaml:86]: Can we put this in the configMap instead?
This part will only exist in the template when encryptedCredentials is not empty

File: charts/zesty/templates/role.yaml
Comment (IdoKendo) [inline charts/zesty/templates/role.yaml:1]: Is there a specific reason as to why we are not using the existing cluster role?

**Review by IdoKendo — CHANGES_REQUESTED** [review]
Hey @bartosz-qubex, I have two questions regarding the changes, please have a look and let me know 🙏

File: charts/zesty/templates/deployment.yaml (hunk 1 above)
Comment (bartosz-qubex) [reply charts/zesty/templates/deployment.yaml:86]: I wanted to use Downward API, which will always point to the correct namespace, but yes. This part will ne be generated when encryptedCredentials is empty. I'll fix that now

File: charts/zesty/templates/role.yaml
Comment (bartosz-qubex) [reply charts/zesty/templates/role.yaml:1]: ClusterRole gives access to whole cluster / all namespaces. By using Role we are limiting permissions only to one namespace

**Pull request merged**